akka-actor-typed/src/main/scala/akka/actor/typed/internal/routing/PoolRouterImpl.scala

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (C) 2009-2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.actor.typed.internal.routing
+
+import akka.actor.typed.Behavior.DeferredBehavior
+import akka.actor.typed._
+import akka.actor.typed.scaladsl.AbstractBehavior
+import akka.actor.typed.scaladsl.ActorContext
+import akka.actor.typed.scaladsl.Behaviors
+import akka.annotation.InternalApi
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[akka] final case class PoolRouterBuilder[T](
+  poolSize:     Int,
+  behavior:     Behavior[T],
+  logicFactory: () ⇒ RoutingLogic[T] = () ⇒ new RoutingLogics.RoundRobinLogic[T]
+) extends DeferredBehavior[T]
```

Comment:
Why is DeferredBehavior needed? I'm comparing with `Behaviors.receive` which returns a `Receive`, which is also like a builder where you can optionally add the signal handler.

Reply:
I wanted to make sure we don't accidentally share the logic, but that a separate instance of logic is used, but it's not strictly necessary, I'll see if I can get rid of it when I get back to it.

Reply:
The difference here is that we need a final build step to create routees etc. while the `Receive` is good to use as a behavior directly. So to avoid the deferred here we'd need to make some explicit `.build` method. I think using `Deferred` is nicer than that.